### PR TITLE
Add Deploy 14.0.1 release notes

### PR DESCRIPTION
**Pull request by ronaldbarendse**
## Description

Release notes for Deploy 14.0.1.

## Type of suggestion

* [ ] Typo/grammar fix
* [ ] Updated outdated content
* [ ] New content
* [x] Updates related to a new version
* [ ] Other

## Product & version (if relevant)

Deploy 14.0.1

## Deadline (if relevant)

Thursday June 6th 🚀

### CODE REVIEW

**Comment by eshanrnh** [comment]
Thanks, @ronaldbarendse for the PR 💪 We will review it as soon as we can 😀

File: 14/umbraco-deploy/release-notes.md [hunk 1]
```diff
@@ -18,6 +18,19 @@ If you are upgrading to a new major version you can find the details about the b
 
 This section contains the release notes for Umbraco Deploy 13 including all changes for this version.
 
+#### [14.0.1](https://github.com/umbraco/Umbraco.Deploy.Issues/issues?q=is%3Aissue+is%3Aclosed+label%3Arelease%2F14.0.1) (June 6th 2024)
+
+* Ensure remote tree uses correct entity type (if multiple entities like folders and items are present in a tree):
+  * `ITransferEntityService.RegisterTransferEntityType(...)` accepts an optional `RemoteTreeDetail` that now exposes the entity type when getting remote entities;
+* Fix `GetMaxRequestLength` endpoint (misaligned HTTP method) preventing file transfers/uploads from local environments;
+* Fix data type deployment due to missing `EditorUiAlias` (requires a schema extraction/data type save to fix);
```
Comment (jonat123) [inline 14/umbraco-deploy/release-notes.md:26]:
```suggestion
* Fix Data Type deployment due to missing `EditorUiAlias` (requires a schema extraction/data type save to fix);
```

File: 14/umbraco-deploy/release-notes.md [hunk 2]
```diff
@@ -18,6 +18,19 @@ If you are upgrading to a new major version you can find the details about the b
...
 
+#### [14.0.1](https://github.com/umbraco/Umbraco.Deploy.Issues/issues?q=is%3Aissue+is%3Aclosed+label%3Arelease%2F14.0.1) (June 6th 2024)
+
+* Ensure remote tree uses correct entity type (if multiple entities like folders and items are present in a tree):
+  * `ITransferEntityService.RegisterTransferEntityType(...)` accepts an optional `RemoteTreeDetail` that now exposes the entity type when getting remote entities;
+* Fix `GetMaxRequestLength` endpoint (misaligned HTTP method) preventing file transfers/uploads from local environments;
+* Fix Data Type deployment due to missing `EditorUiAlias` (requires a schema extraction/Data Type save to fix);
+* Fix JSON serialization errors in trigger endpoints (extract and status report);
+* Obsolete and hide `NestedContentValueConnector` and add import migrators for unsupported legacy editors by default:
+  * Adds `ReplaceMediaPickerDataTypeArtifactMigrator` and `ReplaceNestedContentDataTypeArtifactMigrator` artifact migrators to replace the data types with the Media Picker v3 and Block List respectively;
```
Comment (jonat123) [inline 14/umbraco-deploy/release-notes.md:29]:
```suggestion
  * Adds `ReplaceMediaPickerDataTypeArtifactMigrator` and `ReplaceNestedContentDataTypeArtifactMigrator` artifact migrators to replace the Data Types with the Media Picker v3 and Block List respectively;
```

**Review by jonat123 — APPROVED** [review]
LGTM 💪

**Pull request merged**